lib/elixir_analyzer/exercise_test/assert_call/compiler.ex

```diff
@@ -74,6 +76,15 @@ defmodule ElixirAnalyzer.ExerciseTest.AssertCall.Compiler do
   """
   @spec annotate(Macro.t(), map()) :: {Macro.t(), map()}
   def annotate(node, acc) do
+    imports = track_imports(node) ++ track_aliases(node)
+
+    acc =
+      Enum.reduce(imports, acc, fn {alias, full_path}, acc ->
+        if acc.in_function_def,
+          do: %{acc | in_function_modules: Map.put(acc.in_function_modules, alias, full_path)},
+          else: %{acc | modules_in_scope: Map.put(acc.modules_in_scope, alias, full_path)}
+      end)
```

Comment:
I dont think annotate should be responsible for folding the results of `track_*` into the accumulator, I have included a further code suggestion with passing code below.

```suggestion
    acc =
      acc
      |> track_aliases(node)
      |> track_imports(node)
```